pyomo/gdp/plugins/bigm.py

```diff
@@ -253,117 +224,125 @@ def _apply_to_impl(self, instance, **kwds):
-        # The name used to indicate if thee were OR or XOR disjunctions,
-        # however now that Disjunctions ae allowed to mix the state we
+        # The name used to indicate if there were OR or XOR disjunctions,
+        # however now that Disjunctions are allowed to mix the state we
         # can no longer make that distinction in the name.
         #    nm = '_xor' if xor else '_or'
         nm = '_xor'
         orCname = unique_component_name(
             parent, '_gdp_bigm_relaxation_' + disjunction.getname(
                 fully_qualified=True, name_buffer=NAME_BUFFER) + nm)
         parent.add_component(orCname, orC)
-        orConstraintMap[disjunction] = orC
+        disjunction._algebraic_constraint = weakref_ref(orC)
+
         return orC
 
-    def _transformDisjunction(self, obj, transBlock, bigM):
+    def _transform_disjunction(self, obj, bigM):
+        transBlock = self._add_transformation_block(obj.parent_block())
+
+        # If this is an IndexedDisjunction, we have to create the XOR constraint
+        # here because we want its index to match the disjunction. In any case,
+        # we might as well.
+        xorConstraint = self._get_xor_constraint(obj)
+
         # relax each of the disjunctionDatas
         for i in sorted(iterkeys(obj)):
-            self._transformDisjunctionData(obj[i], transBlock, bigM, i)
+            self._transform_disjunctionData(obj[i], bigM, i, xorConstraint,
+                                            transBlock)
 
-        # deactivate so we know we relaxed
+        # deactivate so the writers don't scream
         obj.deactivate()
 
-    def _transformDisjunctionData(self, obj, transBlock, bigM, index):
+    def _transform_disjunctionData(self, obj, bigM, index, xorConstraint=None,
+                                   transBlock=None):
         if not obj.active:
-            return  # Do not process a deactivated disjunction
-        parent_component = obj.parent_component()
-        transBlock.disjContainers.add(parent_component)
-        orConstraint = self._getXorConstraint(parent_component)
+            return  # Do not process a deactivated disjunction 
+        # We won't have these arguments if this got called straight from
+        # targets. But else, we created them earlier, and have just been passing
+        # them through.
+        if transBlock is None:
+            transBlock = self._add_transformation_block(obj.parent_block())
+        if xorConstraint is None:
+            xorConstraint = self._get_xor_constraint(obj.parent_component())
 
         xor = obj.xor
         or_expr = 0
+        # Just because it's unlikely this is what someone meant to do...    
+        if len(obj.disjuncts) == 0:
+            raise GDP_Error("Disjunction %s is empty. This is " 
+                            "likely indicative of a modeling error."  %
+                            obj.getname(fully_qualified=True,
+                                        name_buffer=NAME_BUFFER))
```

Comment:
Is this possible if someone created an `IndexedDisjunction` and wanted to `Skip` an index?

Reply:
No, because Skip just doesn't create the index, so the DisjunctionData would not be there at all. So I think this is okay.